src/libsyntax/parse/parser.rs

```diff
@@ -2840,9 +2846,13 @@ impl<'a> Parser<'a> {
                 self.parse_prefix_expr(attrs)?
             }
         };
+        let last_type_ascription_set = self.last_type_ascription.is_some();
```

Comment:
Unnecessary condition: in all cases below `self.last_type_ascription = None;` can be done unconditionally.

Reply:
The condition is necessary so that 3027 doesn't erase the assignment in 2940. All other cases don't need it.